pallet-logion-vote/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::codec::{Decode, Encode};
+use frame_support::dispatch::Vec;
+use scale_info::TypeInfo;
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
+pub struct Vote<LocId, AccountId> {
+    loc_id: LocId,
+    ballots: Vec<Ballot<AccountId>>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
+pub struct Ballot<AccountId> {
+    voter: AccountId,
+    status: BallotStatus,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
+pub enum BallotStatus {
+    UnVoted,
+    VotedYes,
+    VotedNo,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+    use codec::HasCompact;
+    use frame_support::{
+        dispatch::DispatchResultWithPostInfo,
+        pallet_prelude::*,
+    };
+    use frame_system::pallet_prelude::OriginFor;
+    use logion_shared::{IsLegalOfficer, LocValidity};
+    use super::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// LOC identifier
+        type LocId: Member + Parameter + Default + Copy + HasCompact;
+
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// Query for checking that a signer is a legal officer
+        type IsLegalOfficer: IsLegalOfficer<Self::AccountId, Self::RuntimeOrigin>;
+
+        /// Query for checking the existence of a closed Identity LOC
+        type LocValidity: LocValidity<Self::LocId, Self::AccountId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub (super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    /// Votes
+    #[pallet::storage]
+    #[pallet::getter(fn votes)]
+    pub type Votes<T> = StorageValue<_, Vec<Vote<<T as Config>::LocId, <T as frame_system::Config>::AccountId>>, ValueQuery>;
```

Comment:
As already discussed, switch to a map and add an additional storage value for the counter.

Reply:
Done.